content/300-guides/050-application-lifecycle/100-developing-with-prisma-migrate/100-advanced-migrate-scenarios.mdx

```diff
@@ -0,0 +1,195 @@
+
+To change the direction of a 1-1 relation:
+
+1. Make the change in the schema:
+
+   ```prisma
+   model User {
+        id      Int      @id @default(autoincrement())
+        name    String
+        posts   Post[]
+        profile Profile?
+        profileId  @relation(fields: profileId, references: id)
+   }
+
+    model Profile {
+        id Int @id @default(autoincrement())
+        biography String
+        user User
+    }
+    ```
+
+1. Run the following command to create a **draft migration** that you can edit before applying to the database:
+
+    <CodeWithResult expanded={true}>
+    <Cmd>
+
+      ```terminal
+      npx prisma migrate dev --name rename-migration --create-only
+      ```
+
+    </Cmd>
+    <CmdResult>
+
+      ```
+      ⚠️  There will be data loss when applying the migration:
+
+      • The migration will add a unique constraint covering the columns `[profileId]` on the table `User`. If there are existing duplicate values, the migration will fail.
+      ```
+
+    </CmdResult>
+    </CodeWithResult>
+
+1. Edit the draft migration as shown:
+
+      <TabbedContent tabs={[<FileWithIcon text="Before" icon="code"/>, <FileWithIcon text="After" icon="code"/>]}>
+
+      <tab>
+
+      ```sql
+      -- DropForeignKey
+      ALTER TABLE "Profile" DROP CONSTRAINT "Profile_userId_fkey";
+
+        -- DropIndex
+      DROP INDEX "Profile_userId_unique";
+
+      -- AlterTable
+      ALTER TABLE "Profile" DROP COLUMN "userId";
+
+      -- AlterTable
+      ALTER TABLE "User" ADD COLUMN     "profileId" INTEGER NOT NULL;
+
+      -- CreateIndex
+      CREATE UNIQUE INDEX "User_profileId_unique" ON "User"("profileId");
+
+      -- AddForeignKey
+      ALTER TABLE "User" ADD FOREIGN KEY ("profileId") REFERENCES "Profile"("id") ON DELETE CASCADE ON UPDATE CASCADE;
+      ```
+
+      </tab>
+
+      <tab>
+
+      ```sql
+      -- DropForeignKey
+      ALTER TABLE "Profile" DROP CONSTRAINT "Profile_userId_fkey";
+
+      -- DropIndex
+      DROP INDEX "Profile_userId_unique";
+
+      -- AlterTable
+      ALTER TABLE "User" ADD COLUMN "profileId" INTEGER;
+
+      UPDATE "User"
+      SET "profileId" = "Profile".id
+      FROM "Profile"
+      WHERE "User".id = "Profile"."userId";
+
+      ALTER TABLE "User" ALTER COLUMN "profileId" SET NOT NULL;
+
+      -- AlterTable
+      ALTER TABLE "Profile" DROP COLUMN "userId";
+
+      -- CreateIndex
+      CREATE UNIQUE INDEX "User_profileId_unique" ON "User"("profileId");
+
+      -- AddForeignKey
+      ALTER TABLE "User" ADD FOREIGN KEY ("profileId") REFERENCES "Profile"("id") ON DELETE CASCADE ON UPDATE CASCADE;
+      ```
+
+      </tab>
+
+      </TabbedContent>
+
+## Populating new required fields with data from existing records
+
+For example, changing name to mandatory `firstName` and `lastName` fields:
```

Comment:
Do you have content for this section? Do you need it? Happy to help with that.

Reply:
@albertoperdomo I don't have anything written at the moment - if you have some ready-made SQL I would be happy to take it! :)